pkg/controllers/cluster/cluster_controller.go

```diff
@@ -103,10 +104,13 @@ func (c *Controller) Start(ctx context.Context) error {
 
 // SetupWithManager creates a controller and register to controller manager.
 func (c *Controller) SetupWithManager(mgr controllerruntime.Manager) error {
-	return utilerrors.NewAggregate([]error{
+	errors := []error{
 		controllerruntime.NewControllerManagedBy(mgr).For(&clusterv1alpha1.Cluster{}).Complete(c),
-		mgr.Add(c),
-	})
+	}
+	if c.ClusterMonitor {
+		errors = append(errors, mgr.Add(c))
+	}
```

Comment:
If `--cluster-monitor==false`, then the `cluster` controller here will not added to `manager`, it equal to `--controllers=-cluster`. Am I right?

Reply:
Not exactly the same. I need  both `--controllers=cluster`  with `--cluster-monitor==false`.

Reply:
Oh, you are right. if `--cluster-monitor==false`, only stopp the `Start` function(monitor).